pkg/controller/awsmachine/awsmachine_controller.go

```diff
@@ -136,133 +138,129 @@ func (r *ReconcileAWSMachine) Reconcile(request reconcile.Request) (reconcile.Re
 
 	// Make sure bootstrap data is available and populated.
 	if scope.Machine.Spec.Bootstrap.Data == nil || *scope.Machine.Spec.Bootstrap.Data == "" {
-		klog.Infof("Waiting for bootstrap data to be available on AWSMachine %q/%q", awsm.Namespace, awsm.Name)
+		scope.Info("Waiting for bootstrap data to be available")
 		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
 	}
 
 	// Call the internal reconciler.
 	if err := r.reconcile(ctx, scope); err != nil {
 		if requeueErr, ok := errors.Cause(err).(capierrors.HasRequeueAfterError); ok {
-			klog.Infof("Reconciliation for AWSMachine %q in namespace %q asked to requeue: %v", awsm.Name, awsm.Namespace, err)
+			scope.Info("Reconciliation asked to requeue", "after", requeueErr.GetRequeueAfter())
 			return reconcile.Result{Requeue: true, RequeueAfter: requeueErr.GetRequeueAfter()}, nil
 		}
+		scope.Error(err, "Failed to reconcile AWSMachine")
 		return reconcile.Result{}, err
 	}
 
+	// TODO(vincepri): Handle deletion and finalizers here.
```

Comment:
Deletion should probably be handled prior to calling `r.reconcile`

Reply:
SGTM, going to follow-up in another PR